src/steamship/agents/service/agent_service.py

```diff
@@ -15,18 +15,90 @@ class AgentService(PackageService):
     """AgentService is a Steamship Package that can use an Agent, Tools, and a provided AgentContext to
     respond to user input."""
 
-    def __init__(self, **kwargs):
+    use_llm_cache: bool
+    """Whether or not to cache LLM calls (for tool selection/direct responses) by default."""
+
+    use_action_cache: bool
+    """Whether or not to cache agent Actions (for tool runs) by default."""
+
+    def __init__(
+        self,
+        use_llm_cache: Optional[bool] = False,
+        use_action_cache: Optional[bool] = False,
+        **kwargs,
+    ):
+        self.use_llm_cache = use_llm_cache
+        self.use_action_cache = use_action_cache
         super().__init__(**kwargs)
 
     ###############################################
     # Tool selection / execution
     ###############################################
 
-    def run_action(self, action: Action, context: AgentContext):
+    def next_action(self, agent: Agent, input_blocks: List[Block], context: AgentContext) -> Action:
+        action: Action = None
+        if context.llm_cache:
+            action = context.llm_cache.lookup(key=input_blocks)
```

Comment:
I think really good idea to log the cache hit as well --- wearing the hat of a developer, it would feel like a positive reward to see that I hit cache (and didn't use tokens). 

Actually.. could be cool if the log had a field for tokens saved!

Reply:
added cache log line.